### PR TITLE
PWGEM/PhotonMeson: activate EMC cut in pairing

### PR DESCRIPTION
**Pull request by dsekihat**
PWGEM/PhotonMeson: activate EMC cut in pairing
Track matching is disabled, until matched track table is ready.

### CODE REVIEW

**Comment by mhemmer-cern** [comment]
Why deactivate the TM for EMCal? In EMCal we already have matched tracks?

**Comment by dsekihat** [comment]
I added a comment in the cut class. In analysis chain, we find a cluster on EMC, then, look for tracks around the cluster. So, it is necessary to have the matched track index in SkimEMCCluster table. Now, indexing is opposite. You can add index column in the SkimEMCCluster table with `DECLARE_SOA_ARRAY_INDEX_COLUMN`. Finally, you can get the matched track with `auto track = cluster.template MatchedTrack_as<T>()` from the cluster.

File: PWGEM/PhotonMeson/Core/EMCPhotonCut.h [hunk 1]
```diff
@@ -46,9 +46,11 @@ class EMCPhotonCut : public TNamed
   static const char* mCutNames[static_cast<int>(EMCPhotonCuts::kNCuts)];
 
   // Temporary function to check if cluster passes selection criteria. To be replaced by framework filters.
-  template <typename Cluster, typename Track>
-  bool IsSelected(Cluster const& cluster, std::optional<Track> const& track = std::nullopt) const
+  template <typename T, typename Cluster>
+  bool IsSelected(Cluster const& cluster) const
```
Comment (mhemmer-cern) [inline PWGEM/PhotonMeson/Core/EMCPhotonCut.h:50]: `typename T` is not needed here anymore right?

File: PWGEM/PhotonMeson/Core/EMCPhotonCut.h [hunk 2]
```diff
@@ -73,10 +75,8 @@ class EMCPhotonCut : public TNamed
   // Temporary function to check if cluster passes a given selection criteria. To be replaced by framework filters.
   // Returns true if a cluster survives the cuts!
   template <typename Cluster, typename Track>
-  bool IsSelectedEMCal(const EMCPhotonCuts& cut, Cluster const& cluster, std::optional<Track> const& track = std::nullopt) const
+  bool IsSelectedEMCal(const EMCPhotonCuts& cut, Cluster const& cluster, Track const& track) const
```
Comment (mhemmer-cern) [inline PWGEM/PhotonMeson/Core/EMCPhotonCut.h:78]: Why change the optional part? I mean we don't always have tracks so I thought having the tracks as optional argument would be good?

File: PWGEM/PhotonMeson/Core/HistogramsLibrary.cxx [hunk 3]
```diff
@@ -147,15 +151,27 @@ void o2::aod::emphotonhistograms::DefineHistograms(THashList* list, const char*
...
+  for (int i = 0; i < nmgg04; i++)
+    mgg04[i] = 0.002 * i;
+
+  const int npTgg10 = 61;
+  float pTgg10[npTgg10] = {};
+  for (int i = 0; i < 50; i++)
+    pTgg10[i] = 0.1 * (i - 0) + 0.0; // from 0 to 5 GeV/c, every 0.1 GeV/c
+  for (int i = 50; i < npTgg10; i++)
+    pTgg10[i] = 0.5 * (i - 50) + 5.0; // from 5 to 10 GeV/c, evety 0.5 GeV/c
+
```
Comment (mhemmer-cern) [inline PWGEM/PhotonMeson/Core/HistogramsLibrary.cxx:165]: do we maybe want to change these later into some configurable or are we sure we don't want to change the binning in the end on this level?

File: PWGEM/PhotonMeson/Core/PHOSPhotonCut.h [hunk 4]
```diff
@@ -41,9 +41,10 @@ class PHOSPhotonCut : public TNamed
   static const char* mCutNames[static_cast<int>(PHOSPhotonCuts::kNCuts)];
 
   // Temporary function to check if track passes selection criteria. To be replaced by framework filters.
-  template <typename T>
-  bool IsSelected(T const& cluster) const
+  template <typename T, typename Cluster>
+  bool IsSelected(Cluster const& cluster) const
```
Comment (mhemmer-cern) [inline PWGEM/PhotonMeson/Core/PHOSPhotonCut.h:45]: same as for EMCal the `typename T` can be removed I think

**Review by mhemmer-cern — COMMENTED** [review]
Just a few comments, other than that I think this looks good. Really great work!

File: PWGEM/PhotonMeson/Core/HistogramsLibrary.cxx (hunk 3 above)
Comment (dsekihat) [reply PWGEM/PhotonMeson/Core/HistogramsLibrary.cxx:165]: this is to reduce histogram size. One can improve this. Personally, I don't like to have variable binning from configurable.

File: PWGEM/PhotonMeson/Core/PHOSPhotonCut.h (hunk 4 above)
Comment (dsekihat) [reply PWGEM/PhotonMeson/Core/PHOSPhotonCut.h:45]: This is necessary to decide object type for track. In addition, IsSelected is called uniformly from PairUtilities.h. Therefore, IsSelected<typename>(cluster or v0) is necessary from all 3 cut classes.

File: PWGEM/PhotonMeson/Core/EMCPhotonCut.h (hunk 2 above)
Comment (mhemmer-cern) [reply PWGEM/PhotonMeson/Core/EMCPhotonCut.h:78]: nah it should be fine, thanks

File: PWGEM/PhotonMeson/Core/HistogramsLibrary.cxx (hunk 3 above)
Comment (mhemmer-cern) [reply PWGEM/PhotonMeson/Core/HistogramsLibrary.cxx:165]: okay 👍

**Review by mhemmer-cern — APPROVED** [review]

**Pull request merged**